src/util.js

```diff
@@ -1,29 +1,47 @@
 'use strict'
 
+// @ts-ignore TODO: switch to cborg
```

Comment:
btw, re cborg, https://github.com/ipld/js-dag-cbor is based on this library, it's very close and the changes made there could be backported fairly easily. It's really just the external API (and code style) that's different. cborg has breaking changes to borc, however, it's much stricter about the data model.
If borc is becoming enough of a problem, let me know and I could do a backport fairly easily and we can drop it from the stack.

Reply:
Ah, cbor.  In various places in our stack we have `cbor`, `borc`, and `cborg`.  From what I can make out `cbor` is the OG which recently split into `cbor` and `cbor-web`, ugh (though weirdly this is [sort of my fault](https://github.com/hildjj/node-cbor/pull/130)).  `borc` forked `cbor` back in the mists of time to address some performance issues but is now only sort of half maintained, `cborg` is our fork of `borc` (perhaps not, IDK, the API is very similar) which is faster still, and stricter.

I want to reduce the number of modules we depend on that do the same thing to decrease the cognitive load of maintenance and also to reduce the bundle size.

I have no strong opinion on which one to use, but if development energy is going into `cborg` and it's faster, and more closely supports the IPLD data model, and we are all about the IPLD data model then that is what we should switch to and stop messing around with the others.  It works well enough in other places where our use-case is quite simple.

I've been switching it out in other parts of the stack but it wasn't a drop-in replacement here so I added the TODO.

> I could do a backport fairly easily and we can drop it from the stack

This would be very helpful, yes please!

Reply:
k, I'll put a backport on my TODO
cborg is a new library, not a fork, and it's been built with strictness and our data model as a design goal, but that does mean it's not as friendly for people that are relaxed about encoding (did you know you can encode `undefined`, `Date` and many other non-IPLD types with js-ipld-dag-cbor and pretend everything's just fine?)

aside: re protons/protobufjs - `@ipld/dag-pb` doesn't even have a dependency, it's all done manually. I did want to get to the same treatment of unixfs decoding as well but that's no longer on the table (as a w3dt project maybe).

Reply:
> did you know you can encode undefined, Date and many other non-IPLD types with js-ipld-dag-cbor and pretend everything's just fine?

🙄 - https://github.com/ipfs/js-ipfs/issues/3043

> aside: re protons/protobufjs - @ipld/dag-pb doesn't even have a dependency, it's all done manually

This is an interesting choice but I feel the discussion of it is outside the scope of this PR.